### PR TITLE
integrating new algo for generating returns for stocks

### PR DESCRIPTION
**Pull request by PySchrat**

### CODE REVIEW

File: models/models.py [hunk 1]
```diff
@@ -43,14 +43,140 @@ def stocks(self):
...
 
-    def simulate(self):
+    def simulate(self,marketVolaPart, sectorVolaPart):
         result = {}
         days = pd.date_range(self.start, self.end, freq=BDay())
         number_days = len(days)
+
+        # # # erik code
+        # build volatility dict
+        volaDict = {'internet': [0.1,0.1], 'semiconductor':[0.1,0.1], 'crazytech': [0.1,0.1]}
```
Comment (shun-liang) [inline models/models.py]: We can just use the volatility of the `Stock` instances.

File: models/models.py [hunk 2]
```diff
@@ -43,14 +43,140 @@ def stocks(self):
...
+
+        # build drift dict
+        driftList = [[x.stock_id, x.mean] for x in self.stocks()]
+        driftDict = {k[0]: k[1] for k in driftList}
```
Comment (shun-liang) [inline models/models.py]: You can skip `driftList` and  use dictionary comprehension directly

```
drift_dict = {stock.stock_id: stock.mean for stock in self.stocks()}
```

File: models/models.py [hunk 3]
```diff
@@ -43,14 +43,140 @@ def stocks(self):
...
+        segementsList = list(self.segments.keys())
+
+        # build a dict which has for every stock its segment as key
+        stockSegments = {}
+        for seg in segementsList:
+            arr = [stock for stock in self.segments[seg].stocks]
+            for stock in arr:
+                stockSegments[stock] = seg
+
+        corrSegments = [[1, 0.8, 0.8], [0.8, 1, 0.8], [0.8, 0.8, 1]]
```
Comment (shun-liang) [inline models/models.py]: Let's represent this as tuples in a CSV file.

File: models/models.py [hunk 4]
```diff
@@ -43,14 +43,140 @@ def stocks(self):
...
+        corrStocks = {'internet': [[1, 0.9], [0.9, 1]], 'semiconductor': [[1, 0.9], [0.9, 1]],
```
Comment (shun-liang) [inline models/models.py]: And here as well.

File: models/models.py [hunk 5]
```diff
@@ -43,14 +43,140 @@ def stocks(self):
...
+
+        # # # erik code
         for stock in self.stocks():
-            result[stock.stock_id] = stock.distribution(scale=stock.volatility, loc=stock.mean, size=number_days)
+            #result[stock.stock_id] = stock.distribution(scale=stock.volatility, loc=stock.mean, size=number_days)
+            result = self.myfunc(volaDict, driftDict, corrStocks, corrSegments, stockSegments,
+                            marketVolaPart, sectorVolaPart,number_days)
         return Report(result, days)
 
+    def myfunc(self, vola, drift, corrStocks, corrSegments, stockSegment, marketVolaPart, sectorVolaPart,length):
```
Comment (shun-liang) [inline models/models.py:80]: Why don't we just pass list of `Segment` instances into this function and then we don't have to form several dictionaries and can then skip many parameters.

File: models/models.py (hunk 3 above)
Comment (PySchrat) [reply models/models.py:69]: Let's change matrix to pandas dataframe where column and row labels are segment names 
-> function for generating gaussion returns needs to  be updated afterwards

File: models/models.py (hunk 4 above)
Comment (PySchrat) [reply models/models.py:70]: Maybe better to change the dict like {'segment' : pd.Dataframe, 'segment2':  pd.Dataframe} like layout for segments
--> also need to change generate return function for this

File: models/models.py (hunk 5 above)
Comment (PySchrat) [reply models/models.py:80]: actually we have all information inside the market class itself, so we can just pass 'self' and marketVolaPart, sectorVolaPart, length

**Pull request closed without merging**